test/fulfillment_order_test.rb

```diff
@@ -139,5 +139,223 @@ def setup
+          :request_body => ActiveSupport::JSON.encode(request_body),
+          :body => ActiveSupport::JSON.encode(fake_response)
+
+        accepted = fulfillment_order.accept_fulfillment_request(message: message)
+
+        assert_equal true, accepted
+        assert_equal 'in_progress', fulfillment_order.status
+      end
+    end
+
+    context "#reject_fulfillment_request" do
+      should "reject a fulfillment request for a fulfillment order" do
+        fulfillment_order = ShopifyAPI::FulfillmentOrder.find(519788021)
+
+        message = "LBTM. Reject this FO fulfillment request"
+        request_body = {
+          'fulfillment_request' => {
+            'message' => message
+          }
+        }
+        fake_response = {
+          fulfillment_order: fulfillment_order.attributes.merge(status: 'open', request_status: 'rejected')
+        }
+        fake "fulfillment_orders/519788021/fulfillment_request/reject", :method => :post,
+          :request_body => ActiveSupport::JSON.encode(request_body),
+          :body => ActiveSupport::JSON.encode(fake_response)
+
+        rejected = fulfillment_order.reject_fulfillment_request(message: message)
+
+        assert_equal true, rejected
+        assert_equal 'open', fulfillment_order.status
+        assert_equal 'rejected', fulfillment_order.request_status
+      end
+    end
+
+    context "#request_cancellation" do
+      should "make a cancellation request for a fulfillment order" do
+        fulfillment_order = ShopifyAPI::FulfillmentOrder.find(519788021)
+
+        message = "Cancelling this please."
+        request_body = {
+          'cancellation_request' => {
+            'message' => message
+          }
+        }
+        cancelling = ActiveSupport::JSON.decode(load_fixture('fulfillment_order'))
+        cancelling['status'] = 'in_progress'
+        cancelling['request_status'] = 'cancellation_accepted'
+        fake "fulfillment_orders/519788021/cancellation_request", :method => :post,
+          :request_body => ActiveSupport::JSON.encode(request_body),
+          :body => ActiveSupport::JSON.encode({ fulfillment_order: cancelling })
+
+        cancelled = fulfillment_order.request_cancellation(message: "Cancelling this please.")
+
+        assert_equal true, cancelled
+        assert_equal 'in_progress', fulfillment_order.status
+        assert_equal 'cancellation_accepted', fulfillment_order.request_status
+      end
+    end
+
+    context "#accept_cancellation_request" do
+      should "accept a cancellation request for a fulfillment order" do
+        fulfillment_order = ShopifyAPI::FulfillmentOrder.find(519788021)
+
+        message = 'Already in-progress. Reject this FO cancellation request'
+        request_body = {
+          'cancellation_request' => {
+            'message' => message
+          }
+        }
+        fake_response = { fulfillment_order: fulfillment_order.attributes.merge(status: 'cancelled') }
```

Comment:
Nit: the request status would also be changed to "cancellation_accepted" as well as the status changed to "cancelled".